worker/src/main/scala/org/apache/celeborn/service/deploy/worker/FetchHandler.scala

```diff
@@ -97,137 +95,142 @@ class FetchHandler(val conf: CelebornConf, val transportConf: TransportConf)
+          handleEndStreamFromClient(bufferStreamEnd)
+        case readAddCredit: PbReadAddCredit =>
+          handleReadAddCredit(readAddCredit.getCredit, readAddCredit.getStreamId)
+        case message: GeneratedMessageV3 =>
+          logError(s"Unknown message $message")
+      }
+    } finally {
+      rpcRequest.body().release()
+    }
+  }
+
+  private def handleLegacyRpcMessage(client: TransportClient, rpcRequest: RpcRequest): Unit = {
+    try {
+      val message = Message.decode(rpcRequest.body().nioByteBuffer())
+      message.`type`() match {
+        case Message.Type.OPEN_STREAM =>
+          val openStream = message.asInstanceOf[OpenStream]
+          handleOpenStreamInternal(
+            client,
+            new String(openStream.shuffleKey, StandardCharsets.UTF_8),
+            new String(openStream.fileName, StandardCharsets.UTF_8),
+            openStream.startMapIndex,
+            openStream.endMapIndex,
+            initialCredit = 0,
+            rpcRequestId = rpcRequest.requestId,
+            isLegacy = true,
+            // legacy [[OpenStream]] doesn't support read local shuffle
+            readLocalShuffle = false)
+        case Message.Type.OPEN_STREAM_WITH_CREDIT =>
+          val openStreamWithCredit = message.asInstanceOf[OpenStreamWithCredit]
+          handleOpenStreamInternal(
+            client,
+            new String(openStreamWithCredit.shuffleKey, StandardCharsets.UTF_8),
+            new String(openStreamWithCredit.fileName, StandardCharsets.UTF_8),
+            openStreamWithCredit.startIndex,
+            openStreamWithCredit.endIndex,
+            openStreamWithCredit.initialCredit,
+            rpcRequestId = rpcRequest.requestId,
+            isLegacy = true,
+            readLocalShuffle = false)
+        case _ =>
+          logError(s"Received an unknown message type id: ${message.`type`.id}")
+      }
+    } catch {
+      case e: Exception =>
+        logError("Catch an error when handle legacy rpc message.", e)
+    }
+
+  }
+
   private def handleOpenStreamInternal(
       client: TransportClient,
       shuffleKey: String,
       fileName: String,
       startIndex: Int,
       endIndex: Int,
       initialCredit: Int,
-      request: RpcRequest,
+      rpcRequestId: Long,
       isLegacy: Boolean,
       readLocalShuffle: Boolean = false): Unit = {
+    workerSource.startTimer(WorkerSource.OPEN_STREAM_TIME, shuffleKey)
     try {
       var fileInfo = getRawFileInfo(shuffleKey, fileName)
       fileInfo.getPartitionType match {
         case PartitionType.REDUCE =>
-          if (endIndex != Integer.MAX_VALUE) {
+          val streamId = chunkStreamManager.nextStreamId()
+          fileInfo = partitionsSorter.getSortedFileInfo(
+            shuffleKey,
+            fileName,
+            fileInfo,
+            startIndex,
+            endIndex)
+          // non-range openStream request
+          if (endIndex == Int.MaxValue && !fileInfo.addStream(streamId)) {
```

Comment:
`getSortedFileInfo` may return the sorted file info even though `endIndex == Int.MaxValue`.  IMO here we can just call original fileInfo's `addStream` like following:
```
      var fileInfo = getRawFileInfo(shuffleKey, fileName)
      fileInfo.getPartitionType match {
        case PartitionType.REDUCE =>
          val streamId = chunkStreamManager.nextStreamId()
          // non-range openStream request
          if (endIndex == Int.MaxValue && !fileInfo.addStream(streamId)) {
            // This branch is entered when the original unsorted file has been deleted by another
            // range's openStream request. retry fetching the sorted FileInfo.
            fileInfo = partitionsSorter.getSortedFileInfo(
              shuffleKey,
              fileName,
              fileInfo,
              startIndex,
              endIndex)
            assert(Utils.isSortedPath(fileInfo.getFilePath))
          }
```

Reply:
we also need to return the sorted file information for range openStream requests.

Reply:
addressed in https://github.com/apache/incubator-celeborn/pull/1932/commits/6c86e0149c161c37aa40a7f96f44f9a44f73bf5d